### DISCUSSION TITLE
How can I load image with WiFi connection only?

### DISCUSSION THREAD

**Discussion by gemiren** [coil-kt/coil#809] [Q&A]
Thanks for the awesome library. I have a case that I need to load image when it's connected to WiFi network only. I don't see this is supported natively in the library. I'd like to extend the HttpFetcher to check for network connection type before making the network call. However, the HttpFetcher class is Internal. Any suggestions?

**Comment by colinrtwhite** [comment]
Your best bet is to use an [`Interceptor`](https://coil-kt.github.io/coil/image_pipeline/#interceptors) to check if wifi is enabled if you're loading an image url. If wifi is disabled, you can fail the request before it's started.

**Reply by gemiren** [reply]
Thanks for the tip. I ended up with using a custom OkHttp client with Interceptor.
